nncf/experimental/torch/nas/bootstrapNAS/elasticity/elastic_width.py

```diff
@@ -0,0 +1,891 @@
+            device = group.elements[0].module.weight.device
+
+            cumulative_filters_importance = torch.zeros(filters_num[0]).to(device)
+            # 1.1 Calculate cumulative importance for all filters in this group
+            for minfo in group.elements:
+                weight = minfo.module.weight
+                if self._weights_normalizer:
+                    weight = self._weights_normalizer(minfo.module.weight)
+                filters_importance = self._filter_importance(weight,
+                                                             minfo.module.target_weight_dim_for_compression)
+                cumulative_filters_importance += filters_importance
+
+            _, reorder_indexes = torch.sort(cumulative_filters_importance, dim=0, descending=True)
+            device = get_model_device(self._target_model)
+            reorder_indexes.to(device)
+            # 1.2 Setup reorder indexes as output mask to reorganize filters
+            for minfo in group.elements:
+                node = self._propagation_graph.get_node_by_id(minfo.nncf_node_id)
+                node.data['output_mask'] = PTNNCFTensor(reorder_indexes)
+
+        # 2. Propagating masks across the graph
+        reorder_algo = FilterReorderingAlgorithm(self._target_model,
+                                                 self._propagation_graph,
+                                                 PT_PRUNING_OPERATOR_METATYPES,
+                                                 PTNNCFPruningTensorProcessor)
+        reorder_algo.reorder_filters()
+
+    def find_pairs_of_nodes_with_different_width(self, pairs_of_nodes: List[Tuple[str, str]]) -> List[int]:
+        """
+        Find pairs of nodes that have different output shapes. It's need to resolve conflict between elastic width and
+        elastic depth. The conflicting situation happens when layers are trimmed and skipped independently.
+        There might be a situation when channels are trimmed for the layers in the beginning of skipped block, and
+        not trimmed in the end, thus violating the necessary condition for skipping a block - output shapes on
+        block boundaries should be the same.
+
+        :param pairs_of_nodes: list of pairs of node names
+        :return: index of nodes pair in the given list that have different output shapes.
+        """
+        pair_indexes = []
+        for idx, (start_node_name, end_node_name) in enumerate(pairs_of_nodes):
+            start_node = self._propagation_graph.get_node_by_name(start_node_name)
+            start_mask = start_node.data['output_mask']
+            end_node = self._propagation_graph.get_node_by_name(end_node_name)
+            end_mask = end_node.data['output_mask']
+
+            start_output_shape = list(set(self._propagation_graph.get_output_shapes_for_node(start_node_name)))
+            end_output_shape = list(set(self._propagation_graph.get_output_shapes_for_node(end_node_name)))
```

Comment:
Could you please explain, why output shapes are wrapping by set firstly and then by list? Set wrapping could change order of shapes dimensions

Reply:
A good catch! Thanks! I've preserved the order using `OrderedDict.from_keys`